src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -823,6 +823,7 @@ func hasDifferentNodeSecret(listener1, listener2 KafkaAPITLS) bool {
 		listener1.NodeSecretRef.Name != listener2.NodeSecretRef.Name
 }
 
+// TODO: might need to change this.
```

Comment:
we should validate:
- cacrt is provided only when mtls is enabled
- the secret exists that's referenced